src/hal/mod.rs

```diff
@@ -22,14 +22,23 @@ and each such struct has a `setup()` method that configures the hardware
 */
 
 
+#[cfg(feature = "mcu_lpc17xx")]
```

Comment:
I mean — test the specific parts of the mcu hal, like doing a `volatilecell` replay verification for uart setup.